ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -281,12 +302,84 @@ static int
 gcp_file_open(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name,
   WT_FS_OPEN_FILE_TYPE file_type, uint32_t flags, WT_FILE_HANDLE **file_handle_ptr)
 {
-    WT_UNUSED(file_system);
     WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(file_type);
-    WT_UNUSED(flags);
-    WT_UNUSED(file_handle_ptr);
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    *file_handle_ptr = nullptr;
+
+    // Google cloud only supports opening the file in read only mode.
+    if ((flags & WT_FS_OPEN_READONLY) == 0 || (flags & WT_FS_OPEN_CREATE) != 0) {
+        std::cerr << "gcp_file_open: read-only access required." << std::endl;
+        return EINVAL;
+    }
+
+    // Only data files and regular files should be opened.
+    if (file_type != WT_FS_OPEN_FILE_TYPE_DATA && file_type != WT_FS_OPEN_FILE_TYPE_REGULAR) {
+        std::cerr << "gcp_file_open: only data file and regular types supported." << std::endl;
+        return EINVAL;
+    }
+
+    // Check if object exists in the cloud.
+    bool exists;
+    size_t size;
+    int ret;
+    if (ret = (fs->gcp_conn->object_exists(name, exists, size) != 0)) {
+        std::cerr << "gcp_file_open: object_exists request to google cloud failed." << std::endl;
+        return ret;
+    }
+    if (!exists) {
+        std::cerr << "gcp_file_open: object named " << name << " does not exist in the bucket."
+                  << std::endl;
+        return EINVAL;
+    }
+
+    // Check if there is already an existing file handle open.
+    auto fh_iterator = std::find_if(fs->fh_list.begin(), fs->fh_list.end(),
+      [name](gcp_file_handle *fh) { return fh->name.compare(name); });
```

Comment:
I believe `fh->name.compare(name)` will return 0 if the names match, should we change this to return a boolean instead? (`fh->name.compare(name) == 0`)

Reply:
addressed here: [9faf10f](https://github.com/wiredtiger/wiredtiger/pull/8777/commits/9faf10fe9b57e6c7c29f5909f2cb8c35c5a3ec8e)